### PR TITLE
Fix local cache identifier to solve the local cache consistent issue

### PR DESCRIPTION
**Pull request by secfree**
### What changes are proposed in this pull request?

Fix local cache identifier

### Why are the changes needed?

If not including `modification time`, it may have consistency issues.

### Does this PR introduce any user facing changes?

NO

### CODE REVIEW

**Comment by secfree** [comment]
cc @beinan

**Comment by humengyu2012** [comment]
https://github.com/prestodb/presto/pull/19746

**Review by maobaolong — APPROVED** [review]
This can be necessary for other compute component other than presto, something like Spark, impala, which can use LocalCacheFileSystem but would not pass through the cache identifier from them without any change.

I think this is a easy way to implements the consistent between LocalCacheFileSystem and underf fileSystem.

@secfree Would you like to update the tiltle of this PR to make it easy to show the purpose of this change. Something like `Fix local cache identifier to solve the consistent issue`

**Comment by secfree** [comment]
> Would you like to update the tiltle of this PR to make it easy to show the purpose of this change.

Updated. Thanks for the advice.

**Comment by maobaolong** [comment]
@huanghua78 Would you like to take a look at this PR, I think this is a most cheap way to implement the consistent between LocalCacheFileSystem and under fileSystem.

File: core/client/hdfs/src/main/java/alluxio/hadoop/LocalCacheFileSystem.java [hunk 1]
```diff
@@ -130,8 +131,15 @@ public FSDataInputStream open(Path path, int bufferSize) throws IOException {
         .setGroup(externalFileStatus.getGroup());
     // FilePath is a unique identifier for a file, however it can be a long string
     // hence using md5 hash of the file path as the identifier in the cache.
-    CacheContext context = CacheContext.defaults().setCacheIdentifier(
-        md5().hashString(externalFileStatus.getPath().toString(), UTF_8).toString());
+    String cacheIdentifier;
```
Comment (huanghua78) [inline core/client/hdfs/src/main/java/alluxio/hadoop/LocalCacheFileSystem.java:134]: Looks good.
Why don't we enable this by default?
I don't see the necessity of introducing another property key.

Comment (secfree) [reply core/client/hdfs/src/main/java/alluxio/hadoop/LocalCacheFileSystem.java:134]: For me, I am not clear that which third-party systems are using this class and if `externalFileStatus.getModificationTime()` has a meaningful value for them.

I got the suggestion from beinan that adding a property key should be better. Maybe @beinan can provide some insights?

Comment (huanghua78) [reply core/client/hdfs/src/main/java/alluxio/hadoop/LocalCacheFileSystem.java:134]: OK.

**Review by huanghua78 — APPROVED** [review]
Looks good.

**Comment by maobaolong** [comment]
alluxio-bot, merge this please

**Comment by maobaolong** [comment]
alluxio-bot, merge this please

**Pull request merged**